katsdpimager/frontend.py

```diff
@@ -643,3 +646,4 @@ def run(args, context, queue, dataset, writer):
             for channel_p in params:
                 process_channel(dataset, args, start_channel, context, queue,
                                 reader, writer, channel_p, array_p, weight_p, subtract_model)
+    writer.finalize()
```

Comment:
Just a python question for the ignorant (i.e. me). Why not make `Writer` a context manager with an` __exit__` method rather than using a `finalize` method as you have done here instead? I always thought that this kind of thing was one of the reasons that the context managers existed.

Reply:
I specifically *don't* want it to finalize if there is an exception. Instead, the product controller will launch a new instance of the job, which will write its own values, and will fail if there are already values in place.